src/test/java/org/example/controllers/ReaderControllerTest.java

```diff
@@ -0,0 +1,174 @@
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.example.util.Util.getErrorResponseFromMvcResult;
+import static org.example.util.Util.getFistReader;
+import static org.example.util.Util.getResponseForIdZeroOrLess;
+import static org.example.util.Util.getResponseForInvalidDecimalId;
+import static org.example.util.Util.getResponseForInvalidFieldsNewReaderDTO;
+import static org.example.util.Util.getTestReaders;
+import static org.example.util.Util.setIdForTestReaders;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+
+@AutoConfigureMockMvc
+@ExtendWith(SpringExtension.class)
+@WebMvcTest(ReaderController.class)
+@Import(DateTimeFormatterConfiguration.class)
+class ReaderControllerTest {
+
+  private static final String REQUEST_PATH = "/api/v1/readers";
+
+  @Autowired
+  MockMvc mvc;
+  @Autowired
+  ObjectMapper objectMapper;
+  @MockBean
+  ReaderService readerService;
+  @Autowired
+  DateTimeFormatter dateTimeFormatter;
+
+  @Test
+  void shouldReturnNotFoundWhenReaderNotPresent() throws Exception {
+    var idTen = 10L;
+    when(readerService.findById(idTen)).thenReturn(Optional.empty());
+    mvc.perform(get(REQUEST_PATH + "/" + idTen)).andExpect(status().isNotFound());
+  }
+
+  @Test
+  void shouldReturnReaderWithId1() throws Exception {
+    var idOne = 1L;
+    when(readerService.findById(idOne)).thenReturn(Optional.of(getFistReader()));
+    var mvcResult = mvc.perform(get(REQUEST_PATH + "/" + idOne)).andExpect(status().isOk())
+        .andReturn();
+
+    var contentAsString = mvcResult.getResponse().getContentAsString();
+    var readerFromResponse = objectMapper.readValue(contentAsString, Reader.class);
+
+    assertAll(() -> assertEquals(idOne, readerFromResponse.getId()),
+        () -> assertEquals("reader", readerFromResponse.getName()));
+
+  }
```

Comment:
The test itself is fine, i'm again about the naming.

1. The test name - it's a bit misleading now, you can think that it's all about id=1. In fact it's more like `shouldSuccessfullyReturnExistingReader`
2. `var existingId = 1L'